tests/test_skewing.py

```diff
@@ -0,0 +1,393 @@
+        op = Operator([Eq(u.forward, u + 1), Eq(v.forward, u.forward.dx2 + v + 1)],
+                      opt=('blocking', 'skewing', {'openmp': True}))
+        op.apply(time_M=0, autotune='basic')
+        assert op._state['autotuning'][0]['runs'] == 12
+        assert op._state['autotuning'][0]['tpr'] == options['squeezer'] + 1
+        assert len(op._state['autotuning'][0]['tuned']) == 5
+
+    def test_hierarchical_skewed_blocking(self):
+        grid = Grid(shape=(64, 64, 64))
+
+        u = TimeFunction(name='u', grid=grid, space_order=2)
+
+        op = Operator(Eq(u.forward, u + 1), opt=('blocking', 'skewing',
+                      {'openmp': False, 'blocklevels': 2}))
+
+        # 'basic' mode
+        op.apply(time_M=0, autotune='basic')
+        assert op._state['autotuning'][0]['runs'] == 10
+        assert op._state['autotuning'][0]['tpr'] == options['squeezer'] + 1
+        assert len(op._state['autotuning'][0]['tuned']) == 4
+
+        # 'aggressive' mode
+        op.apply(time_M=0, autotune='aggressive')
+        assert op._state['autotuning'][1]['runs'] == 38
+        assert op._state['autotuning'][1]['tpr'] == options['squeezer'] + 1
+        assert len(op._state['autotuning'][1]['tuned']) == 4
+
+    @switchconfig(platform='cpu64-dummy')  # To fix the core count
+    def test_multiple_threads_w_skewing(self):
+        """
+        Test autotuning when different ``num_threads`` for a given OpenMP parallel
+        region are attempted.
+        """
+        grid = Grid(shape=(96, 96, 96))
+
+        v = TimeFunction(name='v', grid=grid)
+
+        op = Operator(Eq(v.forward, v + 1), opt=('blocking', 'skewing', {'openmp': True}))
+        op.apply(time_M=0, autotune='max')
+        assert op._state['autotuning'][0]['runs'] == 60  # Would be 30 with `aggressive`
+        assert op._state['autotuning'][0]['tpr'] == options['squeezer'] + 1
+        assert len(op._state['autotuning'][0]['tuned']) == 3
+
+
+class TestDseWithSkewing(object):
+    """
+    This class contains tests mainly inherited from test_dse.py
+    Aims to test interoperability for skewing and prior dse passes
+    """
+    @pytest.mark.parametrize('opt, expected', (['skewing', 2], ['advanced', 2]))
+    def test_time_dependent_split(self, opt, expected):
+        grid = Grid(shape=(10, 10))
+        u = TimeFunction(name='u', grid=grid, time_order=2, space_order=2, save=3)
+        v = TimeFunction(name='v', grid=grid, time_order=2, space_order=0, save=3)
+
+        # The second equation needs a full loop over x/y for u then
+        # a full one over x.y for v
+        eq = [Eq(u.forward, 2 + grid.time_dim),
+              Eq(v.forward, u.forward.dx + u.forward.dy + 1)]
+        op = Operator(eq, opt=opt)
+
+        trees = retrieve_iteration_tree(op)
+        assert len(trees) == expected
+        op()
+
+        assert np.allclose(u.data[2, :, :], 3.0)
+        assert np.allclose(v.data[1, 1:-1, 1:-1], 1.0)
+
+
+class TestDleWithSkewing(object):
+    """
```

Comment:
And same as before, maybe you want to parametrize the the already existing tests in test_dle, not copy-pasting code. 

I appreciate the fact that we said we were going to put everything in test_skewing, but if the price to pay is duplicated code (or nearly duplicated code), thanks but no?

Reply:
Absolutely agree, I can get rid of the duplicate code.